cas-server-support-jpa-ticket-registry/src/main/java/org/jasig/cas/ticket/registry/JpaTicketRegistry.java

```diff
@@ -121,7 +125,8 @@ private Ticket getRawTicket(final String ticketId) {
             if (ticketId.startsWith(TicketGrantingTicket.PREFIX)
                     || ticketId.startsWith(ProxyGrantingTicket.PROXY_GRANTING_TICKET_PREFIX)) {
                 // There is no need to distinguish between TGTs and PGTs since PGTs inherit from TGTs
-                return entityManager.find(TicketGrantingTicketImpl.class, ticketId);
+                return entityManager.find(TicketGrantingTicketImpl.class, ticketId,
+                        lockTgt ? LockModeType.PESSIMISTIC_WRITE : null);
```

Comment:
Understood. Thanks for being thorough. 

So, I suggest we also document this property under the JPA docs, and also capture your exact explanation there too. :+1: from me with doc updates.